astropy/io/ascii/tests/test_ecsv.py

```diff
@@ -553,31 +554,29 @@ def test_round_trip_user_defined_unit(table_cls, tmpdir):
     t['l'] = np.arange(5) * unit
     t.write(filename)
     # without the unit enabled, get UnrecognizedUnit
-    with catch_warnings(u.UnitsWarning) as w:
+    if table_cls is QTable:
+        ctx = pytest.warns(u.UnitsWarning, match=r"'bandpass_sol_lum' did not parse .*")
+    else:
+        ctx = nullcontext()
+    # Note: The read might also generate ResourceWarning, in addition to UnitsWarning
+    with ctx:
         t2 = table_cls.read(filename)
     assert isinstance(t2['l'].unit, u.UnrecognizedUnit)
     assert str(t2['l'].unit) == 'bandpass_sol_lum'
     if table_cls is QTable:
-        assert len(w) == 1
```

Comment:
For whatever reason, `catch_warnings` did not see the `ResourceWarning` but `pytest.warns` did. 🤷